osmhelper/osm_helper.py

```diff
@@ -1,328 +0,0 @@
-                sys.stderr.write(str(routes[master_ref]) + "\n\n")
-
-        OsmHelper.save_routes_to_file(routes)
-
-        return routes
-
-    @staticmethod
-    def get_route_masters(bbox, tags, refresh=False):
-        if refresh:
-            print "Start with fresh route masters"
-            route_masters = {}
-        else:
-            route_masters = OsmHelper.read_route_masters_from_file()
-            # FIXME: when caching we might return too early here and miss routes
-            if len(route_masters) > 1:
-                return route_masters
-
-        api = overpy.Overpass()
-
-        # get all route_master's in bbox area
-        # start out from "tags" relations,
-        # because searching directly for type=route_master does not work with overpass
-        query_str = ('relation(%s)%s;rel(br);(._;);out body;' %
-                     (bbox, tags))
-        result = api.query(query_str)
-
-        for rel in result.get_relations():
-
-            if 'ref' in rel.tags:
-                ref = rel.tags['ref']
-            else:
-                sys.stderr.write(
-                    "RELATION WITHOUT REF: https://www.openstreetmap.org/relation/" + rel.id + "\n")
-                continue
-
-            name = rel.tags['name']
-
-            # use OsmApi library, because overpy does not return members
-            osm_api = OsmApi()
-            routes = OrderedDict()
-            for member in osm_api.RelationGet(rel.id)['member']:
-                # add incomplete route, will be completed later
-                stops = OsmHelper.get_stops_of_route(member['ref'])
-                route = Route(member['ref'], None, None, stops, None, ref)
-                route.add_shape()
-                routes[member['ref']] = route
-
-            route_master = RouteMaster(rel.id, ref, name, routes)
-
-            print route_master
-
-            route_masters[ref] = route_master
-
-        # save route masters for next time
-        with open('data/route_masters.pkl', 'wb') as f:
-            pickle.dump(route_masters, f, pickle.HIGHEST_PROTOCOL)
-
-        return route_masters
-
-    @staticmethod
-    def get_route(routes, route_masters, rel, warn=True):
-        rid = rel.id
-        ref = rel.tags['ref'].replace('B', '')
-        if 'name' in rel.tags:
-            name = rel.tags['name']
-        else:
-            name = "??????"
-
-        if 'from' in rel.tags:
-            fr = rel.tags['from']
-        else:
-            fr = None
-
-        if 'to' in rel.tags:
-            to = rel.tags['to']
-        else:
-            to = None
-
-        # more tags: operator
-
-        print("Adding route %s..." % str(ref))
-
-        if ref in route_masters:
-            # route has a master, so update it in the route_master and add it to the routes
-            stops = route_masters[ref].routes[rid].stops
-            route_masters[ref].routes[rid] = Route(rid, fr, to, stops, route_masters[ref], ref, name)
```

Comment:
Got it! I was actually replacing `Route`(variants) inside a `RouteMaster` when having the same `ref`. This was not correct and has been fixed in the latest commit update. Thank you for spotting me that! Now I'm at 9092 trips.... That are too much? Was the old code not generating all trips? Or is there still a problem?